### ISSUE TITLE
Return statement bodies missing in Printer

### ISSUE THREAD

**Issue by deanm** [haxiomic/haxe-glsl-parser#13]
If I parse/reprint something like

```float det(vec2 a, vec2 b) { return a.x*b.y - b.x*a.y; }```

It comes out as

``` float det(vec2 a,vec2 b){return;}```

**Comment by deanm** [comment]
I think the bug is:

```
class ReturnStatement extends JumpStatement{
	var returnExpression:Expression;
	function new(returnExpression:Expression){
		this.nodeType = ReturnStatementNode(this);
		this.returnExpression = returnExpression;
		super(RETURN);
	}
}
```

Super is called after setting nodeType, so JumpStatement's constructor will overwrite ReturnStatements nodeType to that of a JumpStatement.

```
@:publicFields
class ReturnStatement extends JumpStatement{
	var returnExpression:Expression;
	function new(returnExpression:Expression){
		super(RETURN);
		this.nodeType = ReturnStatementNode(this);
		this.returnExpression = returnExpression;
	}
}
```

Seems to improve the situation for me.  Looks like `StructSpecifier` probably has the same bug.

**Comment by haxiomic** [comment]
Thanks @deanm, look's like you're right - thanks for figuring this one out. I managed to introduce this bug recently when I removed the macro that sets nodeType

**Comment by haxiomic** [comment]
It looks like it affected a good number of classes, fixed it in 79b5f2d921d94b1bbc3c5de60646795ff71aad25
